pkg/inject/sidecar_builder.go

```diff
@@ -81,6 +84,29 @@ func updateEnvMapForEnvoy(vars EnvoyTemplateVariables, env map[string]string, vn
 		// Specify a port value to override the default X-Ray daemon port: 2000
 		env["XRAY_DAEMON_PORT"] = strconv.Itoa(int(vars.XrayDaemonPort))
 
+		// Override the default sampling rate of 0.05 (5%) for AWS X-Ray tracer
+		// The value should be specified as a decimal between 0 and 1.00 (100%)
+		samplingRate, ok := env["XRAY_SAMPLING_RATE"]
+		if ok {
+			// `podAnnotations` contains the sampling rate and gets preference over helm configuration
+			// For now delete this value from env so that we can validate before adding again
+			delete(env, "XRAY_SAMPLING_RATE")
+		} else {
+			// `podAnnotations` doesn't contain the sampling rate so get value from helm configuration
+			samplingRate = vars.XraySamplingRate
+		}
+		if samplingRate != "" {
```

Comment:
Do we need this check ? The samplingRate can be empty if the customer has erroneously specified "" value in pod annotations. In that case, we can notify the Customer of this error ?

Reply:
yes we don't need this check. Now that we implicitly always set this to 0.05 and like you mentioned if the user makes a mistake then we notify via an error.